dotnet/src/Connectors/Connectors.Memory.Milvus/MilvusMemoryStore.cs

```diff
@@ -0,0 +1,429 @@
+        }
+
+        for (int rowNum = 0; rowNum < rowCount; rowNum++)
+        {
+            if (results.Scores[rowNum] >= minRelevanceScore)
+            {
+                yield return (
+                    this.ReadMemoryRecord(data, rowNum, withEmbeddings ? embeddingMap![ids[rowNum]] : null),
+                    results.Scores[rowNum]);
+            }
+        }
+    }
+
+    private MemoryRecord ReadMemoryRecord(IReadOnlyList<FieldData> data, int rowNum, ReadOnlyMemory<float>? externalEmbedding = null)
+    {
+        bool isReference = false;
+        string externalSourceName = string.Empty;
+        string id = string.Empty;
+        string description = string.Empty;
+        string text = string.Empty;
+        string additionalMetadata = string.Empty;
+        ReadOnlyMemory<float>? embedding = null;
+        string key = string.Empty;
+
+        foreach (FieldData field in data)
+        {
+            switch (field.FieldName)
+            {
+                case IsReferenceFieldName when field is FieldData<bool> isReferenceField:
+                    isReference = isReferenceField.Data[rowNum];
+                    break;
+
+                case ExternalSourceNameFieldName when field is FieldData<string> externalSourceNameField:
+                    externalSourceName = externalSourceNameField.Data[rowNum];
+                    break;
+
+                case IdFieldName when field is FieldData<string> idField:
+                    id = idField.Data[rowNum];
+                    break;
+
+                case DescriptionFieldName when field is FieldData<string> descriptionField:
+                    description = descriptionField.Data[rowNum];
+                    break;
+
+                case TextFieldName when field is FieldData<string> textField:
+                    text = textField.Data[rowNum];
+                    break;
+
+                case AdditionalMetadataFieldName when field is FieldData<string> additionalMetadataField:
+                    additionalMetadata = additionalMetadataField.Data[rowNum];
+                    break;
+
+                case EmbeddingFieldName when field is FloatVectorFieldData embeddingField:
+                    Debug.Assert(externalEmbedding is null);
+                    embedding = embeddingField.Data[rowNum];
+                    break;
+
+                case KeyFieldName when field is FieldData<string> keyField:
+                    key = keyField.Data[rowNum];
+                    break;
+
+                case TimestampFieldName:
+                    throw new NotImplementedException();
+
+                default:
+                    continue; // Unknown field - ignore
+            }
+        }
+
+        return new MemoryRecord(
+            new MemoryRecordMetadata(isReference, id, text, description, externalSourceName, additionalMetadata),
+            embedding ?? externalEmbedding ?? Array.Empty<float>(),
+            key,
+            null /* TODO: timestamp */);
```

Comment:
Is this TODO because there's no current mechanism to enable it?

Reply:
Yep - https://github.com/microsoft/semantic-kernel/pull/1941/files#r1313992815. Milvus has no native support for timestamp/datetime types; we could dump it as a string - or as a long (ticks/milliseconds), but I want to check what the Python connector does here first etc.

Reply:
Completed this - the timestamp is now saved as an ISO9601 string.